voting_schemes/electionguard/python-wrapper/src/bulletin_board/electionguard/election.py

```diff
@@ -0,0 +1,123 @@
+        GeopoliticalUnit(
+            object_id="default-geopolitical-unit",
+            name="Default Geopolitical Unit",
+            type=ReportingUnitType.county,
+            contact_information=None,
+        )
+    )
+    m.ballot_styles = [BallotStyle("default-style", ["default-geopolitical-unit"])]
+
+    return m
+
+
+def _contest_id_to_geopolitical_unit_id(contest_id: str) -> str:
+    return f"{contest_id}-geopolitical-unit"
+
+
+def _translate_ballot_styles(
+    styles: List[dict],
+) -> Tuple[List[BallotStyle], List[GeopoliticalUnit]]:
+    """
+    Synthesizes geopolitical units from declared ballot styles.
+    We map each question to a fake geopolitical unit.
+    """
+    ballot_styles: List[BallotStyle] = []
+    unique_contest_ids: Set[str] = set()
+
+    for ballot_style in styles:
+        if (
+            "geopolitical_unit_ids" in ballot_style
+        ):  # explicit geopolitical units, don't override
+            ballot_styles.append(BallotStyle.from_json_object(ballot_style))
+        elif (
+            "contests" in ballot_style
+        ):  # flag that indicates we need to fake some geopolitical units
+            contest_ids = {id for id in ballot_style["contests"]}
+            for id in contest_ids:
+                unique_contest_ids.add(id)
+            ballot_styles.append(
+                BallotStyle.from_json_object(
+                    {
+                        "object_id": ballot_style["object_id"],
+                        "geopolitical_unit_ids": [
+                            _contest_id_to_geopolitical_unit_id(id)
+                            for id in contest_ids
+                        ],
+                    }
+                )
+            )
+
+    return ballot_styles, [
+        GeopoliticalUnit(
+            object_id=_contest_id_to_geopolitical_unit_id(contest_id),
+            name=f"{contest_id} Geopolitical Unit",
+            type=ReportingUnitType.county,
+        )
+        for contest_id in unique_contest_ids
+    ]
+
+
+def parse_description(desc: dict) -> Manifest:
+    ballot_styles, synthetic_geopolitical_units = _translate_ballot_styles(
+        desc.get("ballot_styles", []) or []
+    )
+
+    m = Manifest(
+        election_scope_id="decidim-election",
+        spec_version="v0.95",
+        type=ElectionType.special,
+        start_date=datetime.fromisoformat(desc["start_date"]),
+        end_date=datetime.fromisoformat(desc["end_date"]),
+        geopolitical_units=[
+            GeopoliticalUnit.from_json_object(unit)
+            for unit in (desc.get("geopolitical_units", []) or [])
+        ]
+        + synthetic_geopolitical_units,
+        parties=[],
+        candidates=[
+            Candidate.from_json_object(candidate) for candidate in desc["candidates"]
+        ],
+        contests=[],
+        ballot_styles=ballot_styles,
+        name=InternationalizedText.from_json_object(desc["name"]),
+        contact_information=None,  # TODO: add some
```

Comment:
💬 do we need to add something here or can we delete the TODO?

Reply:
Yep, removed the todo